drivers/overlay/joinleave.go

```diff
@@ -134,5 +136,13 @@ func (d *driver) Leave(nid, eid string) error {
 
 	n.leaveSandbox()
 
+	link, err := netlink.LinkByName(ep.ifName)
+	if err != nil {
+		log.Warnf("Failed to retrieve interface link for interface removal on endpoint leave: %v", err)
```

Comment:
You might have to return here if you can't find the interface. Otherwise it might panic in when you call `LinkDel` with a nil `link`.